std/permissions/mod.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+const { PermissionDenied } = Deno.errors;
+
+function getPermissionString(descriptors: Deno.PermissionDescriptor[]): string {
+  return descriptors
+    .map(pd => {
+      switch (pd.name) {
+        case "read":
+        case "write":
+          return pd.path ? `${pd.name}(${pd.path})` : pd.name;
+        case "net":
+          return pd.url ? `${pd.name}(${pd.url})` : pd.url;
+        default:
+          return pd.name;
+      }
+    })
+    .join(", ");
+}
+
+/** Attempts to grant a set of permissions, resolving with the descriptors of
+ * the permissions that are granted.
+ *
+ *      const perms = await grant({ name: "net" }, { name: "read" });
+ *      if (perms && perms.length === 2) {
+ *        // do something cool that connects to the net and reads files
+ *      } else {
+ *        // notify user of missing permissions
+ *      }
+ *
+ * If one of the permissions requires a prompt, the function will attempt to
+ * prompt for it.  The function resolves with all of the granted permissions. */
+export async function grant(
+  ...descriptors: Deno.PermissionDescriptor[]
+): Promise<void | Deno.PermissionDescriptor[]>;
+/** Attempts to grant a set of permissions, resolving with the descriptors of
+ * the permissions that are granted.
+ *
+ *      const perms = await grant([{ name: "net" }, { name: "read" }]);
+ *      if (perms && perms.length === 2) {
+ *        // do something cool that connects to the net and reads files
+ *      } else {
+ *        // notify user of missing permissions
+ *      }
+ *
+ * If one of the permissions requires a prompt, the function will attempt to
+ * prompt for it.  The function resolves with all of the granted permissions. */
+export async function grant(
+  descriptors: Deno.PermissionDescriptor[]
+): Promise<void | Deno.PermissionDescriptor[]>;
+export async function grant(
+  descriptor: Deno.PermissionDescriptor[] | Deno.PermissionDescriptor,
+  ...descriptors: Deno.PermissionDescriptor[]
+): Promise<void | Deno.PermissionDescriptor[]> {
+  const result: Deno.PermissionDescriptor[] = [];
+  descriptors = Array.isArray(descriptor)
+    ? descriptor
+    : [descriptor, ...descriptors];
+  for (const descriptor of descriptors) {
+    let state = (await Deno.permissions.query(descriptor)).state;
+    if (state === "prompt") {
+      state = (await Deno.permissions.request(descriptor)).state;
+    }
```

Comment:
Is there a difference between this and just `const state = (await Deno.permissions.request(descriptor)).state;`? I believe `request()` handles every situation intuitively.

Reply:
Good point...